cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/SparkCellIterator.java

```diff
@@ -325,7 +329,10 @@ private void maybeRebuildClusteringKeys(@NotNull ByteBuffer columnNameBuf)
         {
             Object newObj = deserialize(field, ByteBufferUtils.extractComponent(columnNameBuf, index++));
             Object oldObj = values[field.position()];
-            if (newRow || oldObj == null || newObj == null || !field.equals(newObj, oldObj))
+            // Historically, we compare equality of clustering keys using the Spark types
+            // to determine if we have moved to a new 'row'. We could also compare using the Cassandra types
+            // or the raw ByteBuffers before converting to Spark types.
+            if (newRow || oldObj == null || newObj == null || !sparkSqlTypeConverter.toSparkType(field.type()).equals(newObj, oldObj))
```

Comment:
Question: is the comment a todo (for moving to compare using Cassandra types/raw bytes)?

Reply:
Not necessarily, comparing with Spark types has worked fine for now. Comparing the raw bytes could be a bit faster. It is just an assumption that was intrinsic but not fully called out.